torch/optim/asgd.py

```diff
@@ -79,7 +79,10 @@ def _init_group(self, group, params_with_grad, grads, mus, axs, etas, state_step
                 # State initialization
                 if len(state) == 0:
                     state["step"] = torch.zeros((), device=p.device, dtype=_get_scalar_dtype())
-                    state["eta"] = torch.tensor(group["lr"], device=p.device, dtype=_get_scalar_dtype())
+                    if isinstance(group["lr"], torch.Tensor):
+                        state["eta"] = group["lr"].clone().detach().to(device=p.device, dtype=_get_scalar_dtype())
```

Comment:
maybe `torch.as_tensor(group["lr"], device=p.device, dtype=_get_scalar_dtype())` could be used instead to handle both cases as it seems to be able to consume both tensors and python scalars